src/main/java/com/wise23/chariteed/service/UserServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.wise23.chariteed.service;
+
+import com.wise23.chariteed.model.User;
+import com.wise23.chariteed.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService, UserDetailsService {
+
+    @Autowired
+    BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public void saveUser(User user) {
+        String encodedPassword = bCryptPasswordEncoder.encode(user.getPassword());
+        user.setPassword(encodedPassword);
+        // user.setRole(Role.USER);
+        userRepository.save(user);
+    }
+
+    @Override
+    public List<Object> isUserPresent(User user) {
+        boolean userExists = false;
+        String message = null;
+        Optional<User> existingUserEmail = userRepository.findByEmail(user.getEmail());
+        if (existingUserEmail.isPresent()) {
+            userExists = true;
+            message = "Email Already Present!";
+        }
+        Optional<User> existingUserMobile = userRepository.findByMobile(user.getMobile());
+        if (existingUserMobile.isPresent()) {
+            userExists = true;
+            message = "Mobile Number Already Present!";
+        }
+        if (existingUserEmail.isPresent() && existingUserMobile.isPresent()) {
+            message = "Email and Mobile Number Both Already Present!";
+        }
+        System.out.println("existingUserEmail.isPresent() - " + existingUserEmail.isPresent()
+                + "existingUserMobile.isPresent() - " + existingUserMobile.isPresent());
+        return Arrays.asList(userExists, message);
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```

Comment:
what is the purpose of this function?

Reply:
in the beginning, we use the email address as the username

Reply:
To get the currently logged in user